language_model.py

```diff
@@ -0,0 +1,69 @@
+from functools import partial
+from multiprocessing import Pool
+from typing import List
+
+import numpy as np
+import scipy.sparse as sp
+from tqdm import tqdm
+
+import utility
+import query_handling
+
+count_vectorizer = sp.load_npz("Generated Files/count_vec_matrix.npz")
+
+doc2word = utility.load_vector_file("Generated Files/doc2word.csv")
+word2vec = utility.load_vector_file("Generated Files/word2vec.csv")
+inverse_w2v = {v: k for k, v in word2vec.items()}
+dirichlet_prior = sum([len(i) for i in list(doc2word.values())]) / len(doc2word)
+
+
+def lm_evaluate_word_doc(word:str, doc, doc_id):
+    word_index = inverse_w2v[word]
+    N_d = len(doc)
+    tf = count_vectorizer[doc_id, word_index]
+    w_freq_in_D = np.sum(count_vectorizer[:, word_index])
+    number_of_word_tokens = len(word2vec)
+    return np.prod(
+        [(N_d / (N_d + dirichlet_prior)),
+         (tf / N_d) + (1 - (N_d / (N_d + dirichlet_prior))),
+         (w_freq_in_D / number_of_word_tokens)])
```

Comment:
Passer udregningen stadig efter der bruges `np.prod`? Før udregnede den vel `(N_d / (N_d + dirichlet_prior)) * (tf / N_d)` og `(1 - (N_d / (N_d + dirichlet_prior))) * (w_freq_in_D / number_of_word_tokens)` og plussede dem sammen, mens jeg tror der nu udregnes `(tf / N_d) + (1 - (N_d / (N_d + dirichlet_prior)))` som ganges med de to andre dele.

Reply:
ja, tror dennis har ret. Det er min fejl. Jeg havde læst formlen anderledes.